src/strategies/moxie/index.ts

```diff
@@ -0,0 +1,190 @@
+        id: true
+      }
+    }
+  };
+
+  const liquidityPoolSubgraphQuery = {
+    userPools: {
+      __args: {
+        where: {
+          user_in: allAddresses,
+        },
+        first: QUERY_LIMIT,
+        skip: 0
+      },
+      totalLPAmount: true,
+      pool: {
+        totalSupply: true,
+        moxieReserve: true,
+      },
+      user: {
+        id: true
+      }
+    }
+  };
+
+  //Adding block to the queries if the snapshot is not latest
+  if (snapshot !== 'latest') {
+    protocolSubgraphQuery.portfolios.__args['block'] = { number: snapshot };
+    liquidityPoolSubgraphQuery.userPools.__args['block'] = { number: snapshot };
+  }
+
+  const fetchSubgraphData = async (subgraphUrl, query, processFunction, key) => {
+    let next_page = 0;
+    while (true) {
+      query[key].__args.skip = next_page * QUERY_LIMIT;
+      const response = await subgraphRequest(subgraphUrl, query);
+      const data = response[Object.keys(response)[0]];
+      
+      processFunction(data);
+
+      if (data.length < QUERY_LIMIT) break;
+      next_page++;
+    }
+  };
+
+  const processProtocolData = (portfolios) => {
+    portfolios.forEach((portfolio) => {
+      const userAddress = getAddress(portfolio.user.id);
+      allAddressesScoreMap[userAddress] += parseFloat(formatUnits(portfolio.unstakedBalance, MOXIE_DECIMALS)) * UNSTAKED_FAN_TOKEN_MULTIPLIER * portfolio.subjectToken.currentPriceInMoxie +
+        parseFloat(formatUnits(portfolio.stakedBalance, MOXIE_DECIMALS)) * STAKED_FAN_TOKEN_MULTIPLIER * portfolio.subjectToken.currentPriceInMoxie;
+    });
+  };
+
+  const processLiquidityPoolData = (userPools) => {
+    userPools.forEach((userPool) => {
+      const userAddress = getAddress(userPool.user.id);
+      allAddressesScoreMap[userAddress] += MOXIE_LIQUIDITY_MULTIPLIER * parseFloat(formatUnits(userPool.totalLPAmount, MOXIE_DECIMALS)) *
+        parseFloat(formatUnits(userPool.pool.moxieReserve, MOXIE_DECIMALS)) /
+        parseFloat(formatUnits(userPool.pool.totalSupply, MOXIE_DECIMALS));
+    });
+  };
+  
+  // RPC Call to get balance of Moxie at a block for users
+  const fetchBalances = async () => {
+    const multi = new Multicaller(network, provider, abi, { blockTag });
+    allAddresses.forEach((address) =>
+      multi.call(address, MOXIE_CONTRACT_ADDRESS, 'balanceOf', [address])
+    );
+    const result: Record<string, BigNumberish> = await multi.execute();
+    Object.entries(result).forEach(([address, balance]) => {
+      let formattedBalance = parseFloat(formatUnits(balance, MOXIE_DECIMALS));
+      allAddressesScoreMap[getAddress(address)] += formattedBalance;
+    });
+  };
+
+  // Fetch data from both subgraphs in parallel
+  await Promise.all([
+    fetchSubgraphData(MOXIE_PROTOCOL_SUBGRAPH_URL, protocolSubgraphQuery, processProtocolData, "portfolios"),
+    fetchSubgraphData(MOXIE_LIQUIDITY_POOL_SUBGRAPH_URL, liquidityPoolSubgraphQuery, processLiquidityPoolData, "userPools"),
+    fetchBalances(),
+  ]);
```

Comment:
It looks like the strategy is crossing the limit of 5 API calls/multicalls https://github.com/snapshot-labs/snapshot-strategies?tab=readme-ov-file#code You see some other way around?

Reply:
Ah I thought the 5 requests included pagination. So would fetching all balances in a single multi-call count as 1 call?

Reply:
Yep